### PR TITLE
Update documentation on resource quota limits

### PR DESCRIPTION
**Pull request by maxsokolovsky**
Related to https://github.com/rancher/rancher/issues/35647

### CODE REVIEW

**Review by catherineluse — APPROVED** [review]
LGTM

**Comment by catherineluse** [comment]
@jtravee There's some more information about this in https://github.com/rancher/dashboard/issues/5016 but it's complicated. To summarize the background context of these changes, I would say:

- Currently in the Rancher UI, if you edit project quota limits or namespace default limits, it will update existing namespaces in the project. This is not desired behavior and the UI is going to change so that only new namespaces are affected by editing resource quotas at the project level, not existing namespaces.
- Currently in the Rancher UI, if you make a project resource quota, then you have namespaces within the project that have quotas adding up to the project quota, there's no resource capacity remaining in the project, so if you try to create a new namespace in the project, you see resource limits set to zero or negative values in the namespace. Because to set the default resource limits for new namespaces in the project, the UI first adds up all the limits for namespaces within the project, then compares that with the project resource quota. If there's any resources left in the project, then the default limit is that remainder or the default namespace quota, whichever is smaller. There are several UI tickets open to try to improve the user experience around this in the UI. https://github.com/rancher/dashboard/issues/5016#issuecomment-1048383086
- The user can create a namespace within a project by using kubectl, only if they add annotations about what project they want to put the namespace in. They can also use annotations to ask for resources to be allocated for the namespace. There is currently an open UI ticket where we want to stop using those annotations to calculate how many resources are left in a project, because those annotations only represent what was requested, not what was actually received. https://github.com/rancher/dashboard/issues/5188

I hope that is enough context. These docs changes are basically capturing how the UI is supposed to work when the bugs are fixed, in an effort to keep the UI and the docs in sync.

File: content/rancher/v2.6/en/project-admin/resource-quotas/quotas-for-projects/_index.md [hunk 1]
```diff
@@ -32,6 +34,24 @@ In the following diagram, a Rancher administrator wants to apply a resource quot
...
 
+Before creating a namespace in a project, Rancher compares the amounts of the project's available resources and requested resources (regardless of whether they come from the default or overriden limits).
+If a new namespace, for any one resource, will exceed the remaining quota for it,
+Rancher will still grant the remaining amount for that resource.
+However, this is not the case with namespaces created outside of Rancher's UI. To namespaces created via `kubectl`, Rancher
```
Comment (btat) [inline content/rancher/v2.6/en/project-admin/resource-quotas/quotas-for-projects/_index.md:40]:
```suggestion
However, this is not the case with namespaces created outside of Rancher's UI. For namespaces created via `kubectl`, Rancher
```

**Review by btat — CHANGES_REQUESTED** [review]
Other than the typo, LGTM.

**Review by btat — APPROVED** [review]

File: content/rancher/v2.6/en/project-admin/resource-quotas/quotas-for-projects/_index.md [hunk 2]
```diff
@@ -32,6 +34,24 @@ In the following diagram, a Rancher administrator wants to apply a resource quot
 
 Let's highlight some more nuanced functionality. If a quota is deleted at the project level, it will also be removed from all namespaces contained within that project, despite any overrides that may exist. Further, updating an existing namespace default limit for a quota at the project level will not result in that value being propagated to existing namespaces in the project; the updated value will only be applied to newly created namespaces in that project. To update a namespace default limit for existing namespaces you can delete and subsequently recreate the quota at the project level with the new default value. This will result in the new default value being applied to all existing namespaces in the project.
 
+Before creating a namespace in a project, Rancher compares the amounts of the project's available resources and requested resources (regardless of whether they come from the default or overriden limits).
+If a new namespace, for any one resource, will exceed the remaining quota for it,
```
Comment (catherineluse) [inline content/rancher/v2.6/en/project-admin/resource-quotas/quotas-for-projects/_index.md:38]: Can we change this sentence to "If the requested resources exceed the remaining capacity in the project for those resources, Rancher will assign the namespace the remaining capacity for that resource."

File: content/rancher/v2.6/en/project-admin/resource-quotas/quotas-for-projects/_index.md [hunk 3]
```diff
@@ -32,6 +34,24 @@ In the following diagram, a Rancher administrator wants to apply a resource quot
 
 Let's highlight some more nuanced functionality. If a quota is deleted at the project level, it will also be removed from all namespaces contained within that project, despite any overrides that may exist. Further, updating an existing namespace default limit for a quota at the project level will not result in that value being propagated to existing namespaces in the project; the updated value will only be applied to newly created namespaces in that project. To update a namespace default limit for existing namespaces you can delete and subsequently recreate the quota at the project level with the new default value. This will result in the new default value being applied to all existing namespaces in the project.
```
Comment (catherineluse) [inline content/rancher/v2.6/en/project-admin/resource-quotas/quotas-for-projects/_index.md:35]: @jtravee this part about

> Further, updating an existing namespace default limit for a quota at the project level will not result in that value being propagated to existing namespaces in the project;

This is the part I was thinking of that is not going to be technically true until this issue is fixed: https://github.com/rancher/dashboard/issues/5188

Can you please open another docs issue that just says this paragraph about the nuanced functionality should be updated for v2.6.5? There are a number of UI issues related to project quotas with most being targeted for v2.6.5. https://github.com/rancher/dashboard/issues?q=is%3Aopen+is%3Aissue+label%3Aarea%2Fproject-quota

**Comment by catherineluse** [comment]
@btat I updated one little thing - i would say it's good to merge now.

File: content/rancher/v2.6/en/project-admin/resource-quotas/quotas-for-projects/_index.md [hunk 4]
```diff
@@ -32,6 +34,24 @@ In the following diagram, a Rancher administrator wants to apply a resource quot
...
 
+Before creating a namespace in a project, Rancher compares the amounts of the project's available resources and requested resources (regardless of whether they come from the default or overriden limits).
+If the requested resources exceed the remaining capacity in the project for those resources, Rancher will assign the namespace the remaining capacity for that resource.
```
Comment (cbron) [inline content/rancher/v2.6/en/project-admin/resource-quotas/quotas-for-projects/_index.md:38]: The paragraph above should start with the fact that this is how UI acts. Having that only on line 40 below is somewhat confusing. Reading just lines 37 and 38 we say "rancher" which intuitively means everything (or from a technical POV its the controllers).

**Pull request merged**